src/ipfsd-daemon.js

```diff
@@ -35,41 +32,36 @@ function translateError (err) {
   throw err
 }
 
+/** @ignore @typedef {import("./index").FactoryOptions} FactoryOptions */
+
 /**
  * ipfsd for a go-ipfs or js-ipfs daemon
  * Create a new node.
  *
- * @class
- * @param {Typedefs.SpawnOptions} [opts]
  */
 class Daemon {
-  constructor (opts = { type: 'go' }) {
-    const rootPath = process.env.testpath
-      ? process.env.testpath
-      : __dirname
-
+  /**
+   * @constructor
+   * @param {FactoryOptions} [opts]
+   */
+  constructor (opts) {
     this.opts = opts
-    const envExec = this.opts.type === 'go' ? process.env.IPFS_GO_EXEC : process.env.IPFS_JS_EXEC
-    this.exec = this.opts.exec || envExec || findIpfsExecutable(this.opts.type, rootPath)
-    this._env = Object.assign({}, process.env, this.opts.env)
-    this.path = this.opts.disposable
-      ? tmpDir(this.opts.type === 'js')
-      : (this.opts.repoPath || defaultRepo(this.opts.type))
-    this.bits = this.opts.initOptions ? this.opts.initOptions.bits : null
-    this.disposable = this.opts.disposable
+    this.path = this.opts.ipfsOptions.repo
+    this.exec = this.opts.ipfsBin
+    this.env = Object.assign({}, this.env, { IPFS_PATH: this.path })
+    this.disposable = opts.disposable
     this.subprocess = null
     this.initialized = fs.existsSync(this.path)
     this.started = false
     this.clean = true
     this.apiAddr = null
     this.gatewayAddr = null
-    /** @member {IpfsClient} */
     this.api = null
   }
 
   setApi (addr) {
     this.apiAddr = multiaddr(addr)
-    this.api = (this.opts.IpfsClient || IpfsClient)(addr)
+    this.api = require(this.opts.ipfsHttp.path)(addr)
```

Comment:
Why has this changed to require a path instead of invoking a constructor?

Reply:
we need both path and ref cause we can't send a ref over http to a remote daemon and the browser can't require from a path this is why the previous version was kinda random about using the correct http-client